server/src/services/form.ts

```diff
@@ -142,17 +164,51 @@ export class DatabaseFormService implements IFormService {
 
   public async addResponse(
     ctx: Context,
+    respondent: string,
     formID: string,
     answers: string[]
-  ): Promise<void> {
-    return; // TODO
+  ): Promise<IResponse> {
+    const response = await Response.create({ respondent: respondent, answers: answers });
+    var query = {'id': form.id};
+    const form = await Form.findOneAndUpdate(query, { $push: {responses: response} });
+    return { id: response.id };
   }
 
   public async addOwner(
     ctx: Context,
     formID: string,
     newOwner: IUser
-  ): Promise<void> {
-    return; // TODO
+  ): Promise<IUser> {
+    var query = {'id': form.id};
+    const form = await Form.findOneAndUpdate(query, { $push: {owners: newOwner} });
+    return { id: newOwner.id };
+  }
+
+  public async addScore(
+    ctx: Context,
+    responseID: string,
+    score: IScore
+  ): Promise<IResponse> {
+    var query = {'id': responseID};
+    const response = await Response.findOneAndUpdate(query, { $push: {scoring.type: score} });
```

Comment:
Should only push the object to `scoring`, not `scoring.type`. I'm not 100% sure you can query with id field like that. I'd probably do something like
```
const response = await Response.findById(responseId);
if (!response) { ... }
response.scoring.push(score);
response.markModified('scoring');
return {id: response.id};
```